src/sql/src/plan/query.rs

```diff
@@ -2911,7 +2911,10 @@ pub fn scalar_type_from_sql(
                         ScalarType::Decimal(precision, scale)
                     }
                     ScalarType::String => {
-                        match name.raw_name().to_string().as_str() {
+                        // TODO(justin): we should look up in the catalog to see
+                        // if this type is actually a length-parameterized
+                        // string.
+                        match name.raw_name().item.as_str() {
```

Comment:
So I promised to refactor `UnresolvedObjectName` to something more useful (an enum instead of a vec of idents) which would also help with this! But it ended up being bigger than I realized... I'll see if I can fit that in on Friday.